### ISSUE TITLE
Clicking on tabsheet tab-bar does not focus the tabsheet

### ISSUE THREAD

**Issue by vaadin-bot** [vaadin/framework#2483]
**Originally by @johndevs**

___
One cannot currently enable keyboard navigation by clicking on the "empty space" on the right of the tabs.

Testcase in sampler:
1. Open http://demo.vaadin.com/sampler#TabSheetKeyboard
2. Close all closable tabs
3. Click somewhere outside of the tabsheet so the focus is moved.
4. Click on the empty space right of the remaining tabs
5. Using the keyboard should work but doesn't.

Tested on FF13 on Linux.

___
 Imported from https://dev.vaadin.com/ issue #8998
